gp2gp-translator/src/main/java/uk/nhs/adaptors/pss/translator/mapper/DiagnosticReportMapper.java

```diff
@@ -0,0 +1,203 @@
+        getCompositionsContainingClusterCompoundStatement(ehrExtract)
+            .stream()
+            .flatMap(ehrComposition -> ehrComposition.getComponent().stream())
+            .map(RCMRMT030101UK04Component4::getCompoundStatement)
+            .filter(Objects::nonNull)
+            .flatMap(compoundStatement -> compoundStatement.getComponent().stream())
+            .map(RCMRMT030101UK04Component02::getNarrativeStatement)
+            .filter(Objects::nonNull)
+            .map(narrativeStatement -> getObservationCommentById(observationComments, narrativeStatement.getId().getRoot()))
+            .flatMap(Optional::stream)
+            .forEach(observationComment -> {
+                observationComment.setEffective(null);
+                observationComment.setComment(getLastLine(observationComment.getComment()));
+            });
+    }
+
+    private List<Reference> getSpecimenReferences(RCMRMT030101UK04CompoundStatement compoundStatement) {
+        return compoundStatement.getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component02::getCompoundStatement)
+            .filter(Objects::nonNull)
+            .filter(ResourceFilterUtil::isSpecimen)
+            .map(compoundStatement1 -> new Reference(new IdType(ResourceType.Specimen.name(), compoundStatement1.getId().get(0).getRoot())))
+            .toList();
+    }
+
+    private void setResultReferences(RCMRMT030101UK04CompoundStatement compoundStatement, DiagnosticReport diagnosticReport) {
+        var resultReferences = compoundStatement.getComponent()
+            .stream()
+            .map(RCMRMT030101UK04Component02::getNarrativeStatement)
+            .filter(Objects::nonNull)
+            .map(narrativeStatement -> new Reference(new IdType(ResourceType.Observation.name(), narrativeStatement.getId().getRoot())))
+            .collect(toCollection(ArrayList::new));
+
+        if (!resultReferences.isEmpty()) {
+            diagnosticReport.setResult(resultReferences);
+        }
+    }
+
+    private Optional<Observation> getObservationCommentById(List<Observation> observationComments, String id) {
+        return observationComments.stream()
+            .filter(observationComment -> id.equals(observationComment.getId()))
+            .findFirst();
+    }
+
+    private Optional<Identifier> createIdentifierExtension(List<II> id) {
+        if (id.size() > 1) {
+            final II idExtension = id.get(1);
+            if (idExtension != null && EXTENSION_IDENTIFIER_ROOT.equals(idExtension.getRoot())) {
+                return Optional.of(new Identifier()
+                    .setSystem(EXTENSION_IDENTIFIER_ROOT)
+                    .setValue(idExtension.getExtension()));
+            }
+        }
+        return Optional.empty();
+    }
+
+    private Optional<Reference> buildContext(List<RCMRMT030101UK04EhrComposition> compositions, List<Encounter> encounters,
```

Comment:
If you implement abstract mapper you can change this to accept a single ehrComposition as it should pass in the current one